### PR TITLE
Add test for NLL: unexpected "free region `` does not outlive" error 

### LINKED ISSUE #49824: NLL: unexpected "free region `` does not outlive" error
In the test `borrowck/borrowck-describe-lvalue.rs`, when using NLL by default for MIR borrowck, I started getting the odd error "Free region does not outlive" error below:

```
    // Field from upvar nested
    {
        let mut x = 0;
           || {
               || { //[mir]~ ERROR free region `` does not outlive
                   let y = &mut x;
                   &mut x; //[ast]~ ERROR cannot borrow `**x` as mutable more than once at a time
                   //[mir]~^ ERROR cannot borrow `x` as mutable more than once at a time
                   *y = 1;
                   drop(y);
                }
           };
    }
```

Seems like a bug.

### PR DESCRIPTION
**Pull request by davidtwco**
Fixes #49824.

r? @pnkfelix @nikomatsakis

### CODE REVIEW

**Comment by pnkfelix** [comment]
@davidtwco Since the test uses `#[rustc_regions]`, which is definitely very specific to the internals of NLL, I think the test should live underneath `src/test/ui/nll/closure-requirements/`.

(that is, I don't want someone who's idly skimming `src/test/ui` to run into something like this.)

**Comment by pnkfelix** [comment]
Actually ... why are you using `#[rustc_regions]` ?

My interpretation of our [request for a test](https://github.com/rust-lang/rust/issues/49824#issuecomment-405559002) is that we want one that reflects the *end user's* experience.

We don't need to be exploring the internals of the region requirements here.

**Comment by davidtwco** [comment]
@pnkfelix I've updated the test to remove `#![rustc_regions]`.

**Comment by pnkfelix** [comment]
@davidtwco so you can tell that I changed my mind about something here.

This test represents a case where NLL could break old code that was accepted by the AST-borrowck.

Its exactly the kind of case where I think the right approach is not to use `#![feature(nll)]`, but instead to make the test encode the behave in the distinct borrow-checkers, so that we can immediately see that this is a scenario where the move to NLL is going to break people's code.

At the very least, I would remove the `#![feature(nll)]` from the test, and add `#[rustc_error]` to the `fn main`. This will force the compiler to error out even when the test is accepted under the AST-borrowck.

File: src/test/ui/issue-49824.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(rustc_attrs)]
```
Comment (pnkfelix) [inline src/test/ui/issue-49824.rs]: please add a comment noting that this test is checking a problem that only arises in NLL mode, and maybe a pointer to the issue-49824.nll.stderr  file, just so someone who looks over these tests in the futures doesn't think that this test is a no-op.

Comment (davidtwco) [reply src/test/ui/issue-49824.rs]: Fixed.

**Pull request merged**